evap/results/tests/test_views.py

```diff
@@ -16,6 +18,18 @@ def setUpTestData(cls):
         mommy.make(UserProfile, username='staff', email="staff@institution.example.com")
 
 
+class TestGetCoursesWithPrefetchedData(TestCase):
+    def test_returns_correct_participant_count(self):
+        """ Regression test for #1248 """
+        course = mommy.make(Course, state='published', _participant_count=2, _voter_count=2)
+        courses = get_courses_with_prefetched_data([course])
+        self.assertEqual(courses[0].num_participants, 2)
+        self.assertEqual(courses[0].num_voters, 2)
+        courses = get_courses_with_prefetched_data(Course.objects.filter(pk=course.pk))
+        self.assertEqual(courses[0].num_participants, 2)
+        self.assertEqual(courses[0].num_voters, 2)
+
```

Comment:
how do you feel about adding another test that is deleting a user who voted for a course and checking that the numbers are still correct?

Reply:
if i understand correctly, that would produce the exactly the situation this test sets up: `_participation_count` and `_voter_count` are set, and the actual number of participant and voters differs from that (it's zero). or wouldn't it?

Reply:
yes, but on a more basic level without assuming what we're doing with the counters when that happens. it's not necessary for me, it just felt like a more thorough test.